lib/cpp/src/thrift/server/TThreadedServer.cpp

```diff
@@ -90,31 +95,71 @@ TThreadedServer::~TThreadedServer() {
 }
 
 void TThreadedServer::serve() {
+  threadFactory_->setDetached(false);
```

Comment:
I am *slightly* concerned about user reusing the thrift factory in other part of the code, so changing this can change the function of their other code. I'm not too sure what the maintainers usually do when there are some minor functional changes.